airgun/widgets.py

```diff
@@ -1118,6 +1118,31 @@ class EditableEntryCheckbox(EditableEntry):
     edit_field = Checkbox(locator=".//input[@type='checkbox']")
 
 
+class EditableEntryMultiCheckbox(EditableEntry):
+    """Should be used in case :class:`EditableEntry` widget represented not by
+    a field, but by a set of checkboxes.
+    """
+
+    def fill(self, value):
+        """Check or unxeck one of the checkboxes
+
+        :param value: string with specification of fields' values
+            Example: value={'details.addons': {'Test addon 1': True, 'Test addon 2': False}}
+        """
+
+        if self.edit_button.is_displayed:
+            self.edit_button.click()
+
+        for addon in value.keys():
+            field_name = addon
+            field_value = value[addon]
+            field = Checkbox(locator=(".//p/text()[normalize-space(.)='{0}']"
```

Comment:
we do not use locators inline, so:
1) we put static locators on a base class level:
```
class EditableEntry(GenericLocatorWidget):
...
    edit_button = Text(".//span[contains(@ng-hide, 'editMode')]")
    edit_field = TextInput(locator=".//*[self::input or self::textarea]")
```
2) or in case of dynamic content, we use init:
```
    def __init__(self, parent, locator=None, name=None, logger=None):
        """Supports initialization via ``locator=`` or ``name=``"""
        if locator and name or not locator and not name:
            raise TypeError('Please specify either locator or name')
        locator = (
            locator or
            ".//dt[normalize-space(.)='{}']"
            "/following-sibling::dd[1]".format(name)
        )
        super(EditableEntry, self).__init__(parent, locator, logger)
```

Reply:
so you can specify template for your locator and then just fill it in a loop, anyway you can play a little to find implementation that is in generic style of the code

Reply:
"Fixing" this adds 17 lines of code and is less eficient. But if it's more consistent with the current style...